src/sort/mod.rs

```diff
@@ -97,22 +97,25 @@ pub trait Sort: Any + Send + Sync + Debug {
         let _ = info;
     }
 
-    /// Extracting an expression (with smallest cost) out of a primitive value
-    fn make_expr(&self, egraph: &EGraph, value: Value) -> (Cost, Expr);
-
-    /// For values like EqSort containers, to make/extract an expression from it
-    /// requires an extractor. Moreover, the extraction may be unsuccessful if
-    /// the extractor is not fully initialized.
-    ///
-    /// The default behavior is to call make_expr
-    fn extract_expr(
+    /// Extracting a term (with smallest cost) out of a primitive value
+    fn extract_term(
         &self,
         egraph: &EGraph,
         value: Value,
         _extractor: &Extractor,
         _termdag: &mut TermDag,
-    ) -> Option<(Cost, Expr)> {
-        Some(self.make_expr(egraph, value))
+    ) -> Option<(Cost, Term)>;
+
+    /// Extracting an expression (with smallest cost) out of a primitive value.
+    /// The default implementation uses [`Sort::extract_term`].
+    fn make_expr(&self, egraph: &EGraph, value: Value) -> (Cost, Expr) {
```

Comment:
Could we just remove this function? I don't believe it's used anymore.

Reply:
It's used here https://github.com/egraphs-good/egglog/blob/main/src/serialize.rs#L315
I can probably inline the implementation however